olp-cpp-sdk-core/src/cache/DefaultCacheImpl.cpp

```diff
@@ -72,16 +72,28 @@ time_t GetRemainingExpiryTime(const std::string& key,
   return expiry;
 }
 
-void PurgeDiskItem(const std::string& key, olp::cache::DiskCache& disk_cache,
+bool PurgeDiskItem(const std::string& key, olp::cache::DiskCache& disk_cache,
                    uint64_t& removed_data_size) {
+  bool result = true;
   auto expiry_key = CreateExpiryKey(key);
   uint64_t data_size = 0u;
 
-  disk_cache.Remove(key, data_size);
+  if (!disk_cache.Remove(key, data_size)) {
+    OLP_SDK_LOG_ERROR_F(kLogTag, "PurgeDiskItem failed to remove key='%s'",
+                        key.c_str());
+    result = false;
+  }
   removed_data_size += data_size;
```

Comment:
Should we put it under if statement?

Reply:
Partially. That's done in the `DiskCache::Remove`. On the other hand not all errors were zeroing `data_size`. Fixed that.